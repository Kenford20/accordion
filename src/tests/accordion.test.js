import React from "react";
import Accordion from "../components/accordion";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "jest-dom/extend-expect";

afterAll(cleanup);
const { container, getByTestId, getAllByTestId } = render(<Accordion />);

it("renders the accordion", () => {
  expect(container).toMatchSnapshot();
});

it("renders a deal header for each deal type", () => {
  const dealHeaders = getAllByTestId("deal-header").map(
    dealHeader => dealHeader.textContent
  );

  expect(dealHeaders).toMatchInlineSnapshot(`
    Array [
      "Local DealsTotal: $60",
      "Getaway DealsTotal: $150",
      "Goods DealsTotal: $240",
      "Groupon DealsTotal: $160",
    ]
  `);
});

it("should display the correct grand total price of all prices", () => {
  const sumOfDealsTotals = getAllByTestId("deals-total")
    .map(dealHeader => dealHeader.textContent.match(/\d+/)[0])
    .reduce((a, b) => +a + +b);

  const grandTotal = getByTestId("grand-total");
  expect(grandTotal).toHaveTextContent(sumOfDealsTotals);
});
