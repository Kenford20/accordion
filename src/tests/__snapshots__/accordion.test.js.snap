// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the accordion 1`] = `
<div>
  <div
    id="accordion"
  >
    <div
      class="deal-header-containers"
      data-testid="deal-header"
    >
      <span>
        Local
         Deals
      </span>
      <span
        data-testid="deals-total"
      >
        Total: $
        60
      </span>
    </div>
    <div
      class="deals-container"
      data-testid="deals-container"
    >
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          1
        </span>
        <span>
          $
          10
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          2
        </span>
        <span>
          $
          20
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          3
        </span>
        <span>
          $
          30
        </span>
      </div>
      <div
        class="add-deal"
      >
        <span
          data-testid="add-deal-btn"
        >
          Add Deal
        </span>
      </div>
    </div>
    <div
      class="deal-header-containers"
      data-testid="deal-header"
    >
      <span>
        Getaway
         Deals
      </span>
      <span
        data-testid="deals-total"
      >
        Total: $
        150
      </span>
    </div>
    <div
      class="deals-container"
      data-testid="deals-container"
    >
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          1
        </span>
        <span>
          $
          40
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          2
        </span>
        <span>
          $
          50
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          3
        </span>
        <span>
          $
          60
        </span>
      </div>
      <div
        class="add-deal"
      >
        <span
          data-testid="add-deal-btn"
        >
          Add Deal
        </span>
      </div>
    </div>
    <div
      class="deal-header-containers"
      data-testid="deal-header"
    >
      <span>
        Goods
         Deals
      </span>
      <span
        data-testid="deals-total"
      >
        Total: $
        240
      </span>
    </div>
    <div
      class="deals-container"
      data-testid="deals-container"
    >
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          1
        </span>
        <span>
          $
          70
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          2
        </span>
        <span>
          $
          80
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          3
        </span>
        <span>
          $
          90
        </span>
      </div>
      <div
        class="add-deal"
      >
        <span
          data-testid="add-deal-btn"
        >
          Add Deal
        </span>
      </div>
    </div>
    <div
      class="deal-header-containers"
      data-testid="deal-header"
    >
      <span>
        Groupon
         Deals
      </span>
      <span
        data-testid="deals-total"
      >
        Total: $
        160
      </span>
    </div>
    <div
      class="deals-container"
      data-testid="deals-container"
    >
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          1
        </span>
        <span>
          $
          150
        </span>
      </div>
      <div
        class="deal"
        data-testid="deal"
      >
        <span>
          Deal 
          2
        </span>
        <span>
          $
          10
        </span>
      </div>
      <div
        class="add-deal"
      >
        <span
          data-testid="add-deal-btn"
        >
          Add Deal
        </span>
      </div>
    </div>
    <div
      id="grand-total"
    >
      <div
        data-testid="grand-total"
      >
        Grand Total: 
        <span>
          $
          610
        </span>
      </div>
    </div>
  </div>
</div>
`;
